scopelint/lint.go

```diff
@@ -42,13 +43,25 @@ func (l *Linter) LintFiles(files map[string][]byte) ([]Problem, error) {
 		} else if strings.TrimSuffix(astFile.Name.Name, "_test") != strings.TrimSuffix(pkgName, "_test") {
 			return nil, fmt.Errorf("%s is in package %s, not %s", filename, astFile.Name.Name, pkgName)
 		}
+
+		info := &types.Info{
+			Defs: make(map[*ast.Ident]types.Object),
+			Uses: make(map[*ast.Ident]types.Object),
+		}
+		conf := types.Config{Importer: importer.Default()}
+		_, err = conf.Check(pkgName, pkg.FileSet, []*ast.File{astFile}, info)
+		if err != nil {
+			return nil, fmt.Errorf("error on get info from %s file in %s package: %w", filename, pkgName, err)
```

Comment:
In this repository, I called it to check itself.
```
go run main.go ./...
```

I got following errors

```
$ go run ./main.go ./...
error on get info from main.go file in main package: main.go:17:2: could not import github.com/kyoh86/scop
elint/scopelint (can't find import: "github.com/kyoh86/scopelint/scopelint")
example/ranges.go:20:10: Using the variable on range scope "key" in function literal
example/ranges.go:20:15: Using the variable on range scope "value" in function literal
example/ranges.go:22:24: Using a reference for the variable on range scope "key"
example/ranges.go:23:28: Using a reference for the variable on range scope "value"
example/ranges.go:38:10: Using the variable on range scope "key" in function literal
example/ranges.go:40:24: Using a reference for the variable on range scope "key"
example/ranges.go:56:10: Using the variable on range scope "i" in function literal
example/ranges.go:57:10: Using the variable on range scope "j" in function literal
example/ranges.go:59:30: Using a reference for the variable on range scope "i"
example/readme.go:10:16: Using the variable on range scope "val" in function literal
example/readme.go:23:28: Using a reference for the variable on range scope "val"
error on get info from scopelint/lint.go file in scopelint package: scopelint/lint.go:19:55: undeclared na
me: Problem
Found 11 lint problems; failing.
exit status 1
```

What is this check?